src/test/java/StringManipulationTest.java

```diff
@@ -89,25 +106,42 @@ public void testGeSubStrings1() {
 
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is my string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 4);
+        assertEquals(sStings[0], "This");
+        assertEquals(sStings[3], "string");
     }
+
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        // test for ignoring leading whitespace
+        manipulatedstring.setString("          This is my string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 4);
+        assertEquals(sStings[0], "This");
+        assertEquals(sStings[3], "string");
     }
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        // test for ignoring whitespace in string
+        manipulatedstring.setString("This     is my       string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 4);  
+        assertEquals(sStings[0], "This");
+        assertEquals(sStings[3], "string");  
     }
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        // test for correctly retrieving one substring in one word
+        manipulatedstring.setString("onlyoneword");
+        String [] sStings = manipulatedstring.getSubStrings(1,1);
+        assertEquals("onlyoneword", sStings[0]);
     }
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        // test for correctly retrieving one substring in multiple word string
+        manipulatedstring.setString("only one substring");
+        String [] sStings = manipulatedstring.getSubStrings(1,1);    
+        assertEquals("only", sStings[0]);
     }
-
     @Test
     public void testRestoreString1()
     {
```

Comment:
Is this suppose to be empty? I believe it should have a test case in it

Reply:
It is not empty - there were just no changes made.